aptos-move/framework/src/natives/cryptography/algebra/hash_to_structure.rs

```diff
@@ -50,30 +50,34 @@ macro_rules! suite_from_ty_arg {
     }};
 }
 
-fn hash_to_bls12381gx_cost(
-    dst_len: usize,
-    msg_len: usize,
-    dst_shortening_base: InternalGas,
-    dst_shortening_per_byte: InternalGasPerByte,
-    mapping_base: InternalGasPerArg,
-    mapping_per_byte: InternalGasPerArg,
-) -> InternalGas {
-    // DST shortening as defined in https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-16.html#name-using-dsts-longer-than-255-.
-    let dst_shortening_cost = if dst_len <= 255 {
-        InternalGas::zero()
-    } else {
-        dst_shortening_base + dst_shortening_per_byte * NumBytes::from((17 + dst_len) as u64)
-    };
+macro_rules! hash_to_bls12381gx_cost {
```

Comment:
Same Q here about the conversion to macro, if the answer is different. I suspect it has something to do with the type of the abstract gas expressions.

Reply:
> I suspect it has something to do with the type of the abstract gas expressions.

Precisely. It's possible to write this as a function, but I found it a bit too verbose for this context since you have to specify the types.